examples/advanced/lr-newton-raphson/README.md

```diff
@@ -0,0 +1,253 @@
+
+
+On the client side, the local training logic is implemented
+[here](job/newton_raphson/app/custom/newton_raphson_train.py). The
+implementation is based on the [`Client
+API`](https://nvflare.readthedocs.io/en/main/programming_guide/execution_api_type.html#client-api). This
+allows user to add minimum `nvflare`-specific codes to turn a typical
+centralized training script to a federated client side local training
+script.
+- During local training, each client receives a copy of the global
+  model, sent by the server, using `flare.receive()` API. The received
+  global model is an instance of `FLModel`.
+- A local validation is first performed, where validation metrics
+  (accuracy and precision) are streamed to server using the
+  [`SummaryWriter`](https://nvflare.readthedocs.io/en/main/apidocs/nvflare.client.tracking.html#nvflare.client.tracking.SummaryWriter). The
+  streamed metrics can be loaded and visualized using tensorboard.
+- Then each client computes it's gradient and Hessian based on local
+  training data, using their respective theoretical formula described
+  above. This is implemented in the
+  [`train_newton_raphson()`](job/newton_raphson/app/custom/newton_raphson_train.py#L82)
+  method. Each client then sends the computed results (always in
+  `FLModel` format) to server for aggregation, using `flare.send()`
+  API.
+
+Each client site corresponds to a site listed in the data table above.
+
+A [centralized training script](./train_centralized.py) is also
+provided, which allows for comparing the federated Newton-Raphson
+optimization versus the centralized version. In the centralized
+version, training data samples from all 4 sites were concatenated into
+a single matrix, used to optimize the model parameters. The
+optimized model was then tested separately on testing data samples of
+the 4 sites, using accuracy and precision as metrics.
+
+Comparing the federated [client-side training
+code](job/newton_raphson/app/custom/newton_raphson_train.py) with the
+centralized [training code](./train_centralized.py), we can see that
+the training logic remains similar: load data, perform training
+(Newton-Raphson updates), and valid trained model. The only added
+differences in the federated code are related to interaction with the
+FL system, such as receiving and send `FLModel`.
+
+## Set Up Environment & Install Dependencies
+
+Create & activate a virtual environment
+```
+virtualenv -p python3 ./venv
+source ./venv/bin/activate
+```
+
+Install dependencies
+```
+pip install -r requirements.txt
+```
+This will install main dev branch of `NVFlare` and `FLamby`
+
+## Download and prepare data
+
+Execute the following script
+```
+bash ./prepare_heart_disease_data.sh
+```
+This will download the heart disease dataset under
+`/tmp/flare/dataset/heart_disease_data/`
+
+## Centralized Logistic Regression
+
+Launch the following script:
+```
+./train_centralized.py --solver custom
```

Comment:
You need to add `#!/usr/bin/env python` to your script or update the command here to include `python ...` for this to work.